homeassistant/components/apsystemsapi_local/__init__.py

```diff
@@ -0,0 +1,48 @@
+"""The APsystems local API integration."""
+
+from __future__ import annotations
+
+import logging
+
+from APsystemsEZ1 import APsystemsEZ1M
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN, UPDATE_INTERVAL
+from .coordinator import ApSystemsDataCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
+    hass.data.setdefault(DOMAIN, {})
+    # https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities
+
+    api = APsystemsEZ1M(ip_address=entry.data[CONF_IP_ADDRESS], timeout=8)
+    coordinator = ApSystemsDataCoordinator(
+        hass, api, interval=entry.data.get(UPDATE_INTERVAL)
+    )
+    hass.data[DOMAIN][entry.entry_id] = {**entry.data, "COORDINATOR": coordinator}
```

Comment:
Do a first refresh so you know the connection works
```suggestion
    await coordinator.async_config_entry_first_refresh()
    hass.data[DOMAIN][entry.entry_id] = {**entry.data, "COORDINATOR": coordinator}
```